driver/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java

```diff
@@ -155,11 +169,29 @@ private void checkClosed() {
         }
     }
 
-    private Document getChunk(final int chunkIndexToFetch) {
-        return chunksCollection.find(new Document("files_id", fileId).append("n", chunkIndexToFetch)).first();
+    private Document getChunk(final int startChunkIndex) {
+        if (cursor == null) {
+            cursor = chunksCollection.find(new Document("files_id", fileId).append("n", new Document("$gte", startChunkIndex)))
+                    .batchSize(batchSize).sort(new Document("n", 1)).iterator();
+        }
+        Document chunk = null;
+        if (cursor.hasNext()) {
+            chunk = cursor.next();
+            if (batchSize == 1) {
```

Comment:
Few questions about this bit of code: 

* Why the check for batch size here?  Is that because of the "special" meaning of batchSize of 1?
* Should it be responsible for validating that the "n" field equals chunkIndexToFetch?  I see that's done in validateData, but that check seems to belong here instead to ensure the method contract is not violated.